go-runtime/compile/build.go

```diff
@@ -123,11 +136,13 @@ func Build(ctx context.Context, moduleDir string, sch *schema.Schema, filesTrans
 
 	buildDir := buildDir(moduleDir)
 	logger.Debugf("Extracting schema")
-	nativeNames, main, schemaErrs, err := ExtractModuleSchema(moduleDir)
+	parseResult, err := ExtractModuleSchema(moduleDir)
 	if err != nil {
 		return fmt.Errorf("failed to extract module schema: %w", err)
 	}
-	if len(schemaErrs) > 0 {
+	pr := parseResult.MustGet()
```

Comment:
This is usually a code smell. If we're guaranteed that there's a schema, it shouldn't be an optional, and if it is an optional we should be checking for its presence?

Reply:
if err is not nil, it will be empty, but if err is nil we expect a parse result - so this is kind of inferred from advancing beyond the prior `err != nil` check. we could add an explicit assertion though to be safe